File: 55-jump-game/jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int n=nums.length;
        int last_index=n-1;

        for(int i=n-2; i>=0 ;i--){
            if(nums[i]>=last_index-i){
                last_index=i;
            }
        }

        return last_index==0;
    }
}

File: 55-jump-game/jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n= len(nums)
        last_index=n-1
        for i in range(n-2,-1,-1):
            if nums[i]>=last_index-i:
                last_index=i
            
        return last_index==0